lib/pacemaker/pcmk_injections.c

```diff
@@ -89,30 +90,46 @@ inject_transient_attr(pcmk__output_t *out, xmlNode *cib_node,
  *                             7 when interval_ms is 0, inject nothing)
  */
 void
-pcmk__inject_failcount(pcmk__output_t *out, xmlNode *cib_node,
+pcmk__inject_failcount(pcmk__output_t *out, cib_t *cib_conn, xmlNode *cib_node,
                        const char *resource, const char *task,
                        guint interval_ms, int rc)
 {
-    if (rc == 0) {
-        return;
+    char *name = NULL;
+    char *value = NULL;
 
-    } else if ((rc == 7) && (interval_ms == 0)) {
-        return;
+    int failcount = 0;
+    xmlNode *output = NULL;
 
-    } else {
-        char *name = NULL;
-        char *now = pcmk__ttoa(time(NULL));
+    CRM_CHECK(cib_node != NULL, return);
+
+    if ((rc == 0) || ((rc == 7) && (interval_ms == 0))) {
+        return;
+    }
 
-        name = pcmk__failcount_name(resource, task, interval_ms);
-        inject_transient_attr(out, cib_node, name, "value++");
-        free(name);
+    // Get current failcount and increment it
+    name = pcmk__failcount_name(resource, task, interval_ms);
 
-        name = pcmk__lastfailure_name(resource, task, interval_ms);
-        inject_transient_attr(out, cib_node, name, now);
-        free(name);
+    if (cib__get_node_attrs(out, cib_conn, PCMK_XE_STATUS,
+                            pcmk__xe_id(cib_node), NULL, NULL, NULL, name,
+                            NULL, &output) == pcmk_rc_ok) {
 
-        free(now);
+        if (crm_element_value_int(output, name, &failcount) != 0) {
+            failcount = 0;
+        }
     }
+    value = crm_strdup_printf("%d", failcount + 1);
```

Comment:
or equivalently `pcmk__itoa(failcount + 1)`

Reply:
Oh that's right